crates/core_simd/src/vector.rs

```diff
@@ -111,12 +111,42 @@ where
         idxs: Simd<usize, LANES>,
         or: Self,
     ) -> Self {
-        let mask = (mask & idxs.lanes_lt(Simd::splat(slice.len()))).to_int();
+        let mask: Mask<isize, LANES> = mask & idxs.lanes_lt(Simd::splat(slice.len()));
+        // SAFETY: We have masked all the out-of-bounds lanes off.
+        unsafe { Self::gather_select_unchecked(slice, mask, idxs, or) }
+    }
+
+    /// Unsafe SIMD gather: construct a SIMD vector by reading from a slice, using potentially discontiguous indices.
```

Comment:
I would change this doc to something more like the rest of std (with a regular description of the function, and a Safety heading with the invariants below)

Reply:
Can you be more specific what function you are looking at as an example? std has... diverse standards of documentation in actuality.

Reply:
Just a random example: https://doc.rust-lang.org/std/boxed/struct.Box.html#method.from_raw

The beginning "Unsafe SIMD gather" seems repetitive, since that's pretty much what the function signature tells you.  I think the first line should basically say _what_ a gather does, and then go from there.  I think this and `gather_select` should basically have the same one line description, but this should then describe what must be done to use this correctly, but `gather_select` should instead say what causes it to panic.

Reply:
> Collect discontiguous slice elements into a vector.
>
> Copy elements from `slice`, indexed by `idxs`, into a vector, skipping any lanes disabled by `mask`.
>
> # Safety
> Calling this function with out-of-bounds, enabled indices is undefined behavior.

I'm thinking something like this format, but written more clearly

Reply:
we should probably go over our existing docs at some point and make sure they're consistent and make sense...

Reply:
fwiw, if you want to commit this now and deal with docs later, I'm fine with that too. the implementation looks good.

Reply:
I want to commit but I also want stuff in good condition. 😅
Docs should probably be a bar, going forward, and **especially** for `unsafe fn`.

Reply:
Applied an attempt at this.